### ISSUE TITLE
Feature Request: floating point precision

### ISSUE THREAD

**Issue by njaard** [maciejhirsz/json-rust#140]
When I dump my json object, I would like to control how much precision it outputs with. This is useful because as a transit format, I don't want huge payloads because I'm printing 12 digits of my floats.

Maybe a good way to implement this is supporting decimal real numbers from some other crate.

**Comment by njaard** [comment]
This is also important for sending stuff to Elasticsearch, which doesn't accept floating point values without a decimal point.

**Comment by maciejhirsz** [comment]
> This is also important for sending stuff to Elasticsearch, which doesn't accept floating point values without a decimal point.

You mean it doesn't accept `10` as `10.0`? How do they handle JSON produced by JavaScript builtins?

**Comment by njaard** [comment]
> You mean it doesn't accept 10 as 10.0?

Yes, that's what I mean. If ES decides that a field a floating point value, it won't accept things that look like integers in that field.

> How do they handle JSON produced by JavaScript builtins?

I don't know
